packages/cdk8s/lib/app.ts

```diff
@@ -36,8 +40,32 @@ export class App extends Construct {
       fs.mkdirSync(this.outdir);
     }
 
-    Node.of(this).synthesize({
-      outdir: this.outdir
-    });
+    // create explicit chart dependencies
+    // from implicit construct dependencies
+    for (const dep of Node.of(this).dependencies) {
+
+      const sourceChart = Chart.of(dep.source);
+      const targetChart = Chart.of(dep.target);
+
+      if (sourceChart !== targetChart) {
+        Node.of(sourceChart).addDependency(targetChart);
+      }
+    }
+
+    // TODO add validate call here
+
+    const charts: IConstruct[] = new DependencyGraph(Node.of(this)).topology().filter(x => x instanceof Chart);
+
+    for (const index in charts) {
```

Comment:
It's a bit frowned upon to use `for-in`. I would just create a separate index:

```ts
let index = 0;
for (const chart of charts) {
  // do your magic.
  index++;
}
```

Reply:
👍